src/pybind/mgr/cephadm/services/iscsi.py

```diff
@@ -76,27 +76,37 @@ def config_dashboard(self, daemon_descrs: List[DaemonDescription]) -> None:
         def get_set_cmd_dicts(out: str) -> List[dict]:
             gateways = json.loads(out)['gateways']
             cmd_dicts = []
+            spec = cast(IscsiServiceSpec,
+                        self.mgr.spec_store.specs.get(daemon_descrs[0].service_name(), None))
+            if spec.api_secure and spec.ssl_cert and spec.ssl_key:
+                cmd_dicts.append({
+                    'prefix': 'dashboard set-iscsi-api-ssl-verification',
+                    'value': "false"
+                })
+            else:
+                cmd_dicts.append({
+                    'prefix': 'dashboard set-iscsi-api-ssl-verification',
+                    'value': "true"
+                })
             for dd in daemon_descrs:
                 spec = cast(IscsiServiceSpec,
                             self.mgr.spec_store.specs.get(dd.service_name(), None))
                 if not spec:
                     logger.warning('No ServiceSpec found for %s', dd)
                     continue
-                if not all([spec.api_user, spec.api_password]):
-                    reason = 'api_user or api_password is not specified in ServiceSpec'
-                    logger.warning(
-                        'Unable to add iSCSI gateway to the Dashboard for %s: %s', dd, reason)
-                    continue
-                host = self._inventory_get_addr(dd.hostname)
-                service_url = 'http://{}:{}@{}:{}'.format(
-                    spec.api_user, spec.api_password, host, spec.api_port or '5000')
-                gw = gateways.get(host)
+                ip = utils.resolve_ip(dd.hostname)
+                protocol = "http"
+                if spec.api_secure and spec.ssl_cert and spec.ssl_key:
+                    protocol = "https"
+                service_url = '{}://{}:{}@{}:{}'.format(
+                    protocol, spec.api_user, spec.api_password, ip, spec.api_port or '5000')
+                gw = gateways.get(dd.hostname)
                 if not gw or gw['service_url'] != service_url:
                     logger.info('Adding iSCSI gateway %s to Dashboard', service_url)
```

Comment:
I think we should avoid to show users and passwords in the log file:
```
safe_service_url = '{}://{}:{}@{}:{}'.format(
                host = self._inventory_get_addr(dd.hostname)
                           protocol, '******', '******', ip, spec.api_port or '5000')
logger.info('Adding iSCSI gateway %s to Dashboard', safe_service_url)
```

Reply:
I  tried making a change of this style and it came out like this 

![image](https://user-images.githubusercontent.com/47704447/101199607-252efb00-3633-11eb-8f25-20945fb96477.png)

As you can see, the safe url was printed during the initial message at the top, but in the last line of the screenshot you can see the url with the actual user/password was still printed. That's because it is printed here https://github.com/ceph/ceph/blob/master/src/pybind/mgr/cephadm/services/cephadmservice.py#L200 
which outputs the actual input we are passing to the mon command (the actual service url). Thoughts? Maybe we want an extra optional parameter to _check_and_set_dashboard (bool, defaulting to True) to say if if it's okay to print out the command contents?

Reply:
Right. we shouldn't print the password to the log.

Reply:
added the "safe_service_url" when printing to the log in the iscsi service file and removed the general printout of the mon command in _check_and_set_dashboard as discussed in standup.